blobmodel/blobs.py

```diff
@@ -48,8 +48,14 @@ def __x_shape(self, x, t):
             raise NotImplementedError(
                 self.__class__.__name__ + '.blob shape not implemented')
 
-    def __y_shape(self, y, t):
-        return np.exp(-((y-self.__get_y_blob_pos(t))**2/(2*self.width_y**2)))
+    def __y_shape(self, y, t, periodic_y, Ly):
+        centered_y_diffs = y
+        if periodic_y:
+            # The y_diff is centered in the simulation domain, if the difference is larger than half the domain,
+            # the previous is used.
+            centered_y_diffs = (y - self.__get_y_blob_pos(t)) % Ly
+            centered_y_diffs[centered_y_diffs > np.max(y)/2] -= Ly
+        return np.exp(-(centered_y_diffs**2/(2*self.width_y**2)))
```

Comment:
Line 58 can't be right
Try examples/show_animation.py with periodic_y=False as test

Reply:
Indeed, good catch. Should be fixed in my next commit.